app/src/main/java/com/example/rwothoromo/developers/view/MainActivity.java

```diff
@@ -116,4 +147,28 @@ public void showGithubUsers(List<GithubUser> githubUsers) {
 		RecyclerView.Adapter adapter = new GithubAdapter(githubUsers, this);
 		recyclerView.setAdapter(adapter);
 	}
+
+	/**
+	 * Retrieve GitHub API data
+	 */
+	public void retrieveData() {
+
+		customProgressDialog(MainActivity.this,
+				"Fetching data from the GitHub API...");
+
+		GithubPresenter githubPresenter = new GithubPresenter(this);
```

Comment:
Do you really always have to reinitialise the presenter every time you call this function? You could make it a field instead and initialise it once in `onCreate`

Reply:
I'll move it to `onCreate`